models/comments.models.js

```diff
@@ -1,31 +1,74 @@
-const db = require("../db/connection")
+const db = require("../db/connection");
 
 exports.selectComments = (review_id) => {
-    return db
+  return db
     .query(
-        `
+      `
         SELECT *
         FROM reviews
         WHERE review_id = $1;
-        `, [review_id]
+        `,
+      [review_id]
     )
     .then(({ rows }) => {
-    if (rows.length === 0){
-        return Promise.reject({status: 404, message: "Review_id not found"})
-    } else {
-        return db
-        .query(
-            `
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, message: "Review_id not found" });
+      } else {
+        return db.query(
+          `
             SELECT *
             FROM comments
             WHERE review_id = $1
             ORDER BY created_at DESC
-            `, [review_id]
-        )
-        .then(({rows}) =>{
-            return rows
-        })
-    }
-        
-    } 
-)};
\ No newline at end of file
+            `,
+          [review_id]
+        );
+      }
+    })
+    .then(({ rows }) => {
+      return rows;
+    });
+};
+
+exports.insertComment = (review_id, username, body) => {
+  return db
+    .query(
+      `
+      SELECT *
+      FROM users
+      WHERE username = $1;
+    `,
+      [username]
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, message: "Username not found!" });
+      } else {
+        return db.query(
+          `
+          SELECT *
+          FROM reviews
+          WHERE review_id = $1
+          `,
+          [review_id]
+        );
+      }
+    })
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, message: "Review_id not found" });
+      } else {
+        return db.query(
+          `
+        INSERT INTO comments (body, author, review_id)
+        VALUES ($1,$2,$3)
+        RETURNING *;
+      `,
+          [body, username, review_id]
+        );
+      }
+    })
+    .then(({ rows }) => {
+      return rows;
+    });
+};
```

Comment:
We dont actually need to do both the user and the review query here since it will be quite expensive if we are dealing with many many requests. One or two requests is fine, but 1 thousand request means 3 thousand queries. Im happy for you to merge since the code does work and i'd like you to get stuck in on 8 more, but ideally we would just do this in a single query. When we insert a comment, SQL will check if the user and the article_id exists in their respective tables since they are Foreign Keys. If they dont exist then it will throw an error, so it will actually do both of these checks for you and reject if either of them dont exist.